awsmp/models.py

```diff
@@ -764,6 +764,51 @@ def _convert_terms_to_dict(self) -> dict[str, Any]:
 
         return yaml_config
 
+    @staticmethod
+    def _mask_unchecked_fields(yaml_config: dict[str, Any]) -> dict[str, Any]:
+        """
+        Insert stub values for fields that are itrnoed during diffing.
+
+        The `diff` command skips the `version` and `eula_docuemnt` keys,
+        but the data mdel still checks those fields. This helper
+        put placeholder data for those keys so the config passes validation
+        without changing the real local config file.
+
+        :param dict[str, Any] yaml_config: dictionary data from loading local yaml config file
+        :return: dictionary data with stub values in unchecked fields
+        :rtype: dict[str, Any]
+        """
+
+        if "version" in yaml_config["product"]:
+            yaml_config["product"].pop("version")
+        if "eula_document" in yaml_config["offer"]:
+            yaml_config["offer"].pop("eula_document")
+
+        stub_value = {
+            "eula_document": [{"type": "CustomEula", "url": "https://temp.com"}],
+            "version": {
+                "version_title": "temp",
+                "release_notes": "temp",
+                "access_role_arn": "arn:aws:iam::123456789123",
+                "usage_instructions": "temp",
+                "recommended_instance_type": "m5.large",
+                "ip_protocol": "tcp",
+                "ip_ranges": ["0.0.0.0/0"],
+                "from_port": 22,
+                "to_port": 22,
+                "ami_id": "ami-12345678912345678",
+                "os_system_name": "temp",
+                "os_user_name": "temp",
+                "os_system_version": "temp-version",
```

Comment:
replacing "temp" with "stub_<field>" would be nice for clarity. 

for arn, i would do the same thing after the `::` unless we do explicit string validation on that field (which we may want to in the future, so not hard set on this one)